tests/test_regularization.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+import random
+
+import numpy as np
+import pandas as pd
+import pytest
+import torch
+from utils.dataset_generators import generate_event_dataset, generate_holiday_dataset
+
+from neuralprophet import NeuralProphet, df_utils
+from neuralprophet.utils import reg_func_abs
+
+# Fix random seeds
+torch.manual_seed(0)
+random.seed(0)
+np.random.seed(0)
+
+# Variables
+EPOCHS = 10
+BATCH_SIZE_HOLIDAYS = 32
+BATCH_SIZE_EVENTS = 1
+LEARNING_RATE = 0.1
+REGULARIZATION = 10
+# Map holiday name to a y value for dataset generation
+Y_HOLIDAYS_OVERRIDE = {
+    "Washington's Birthday": 1,
+    "Labor Day": 5,
```

Comment:
for simplicity it may be better to keep all override values the same.
I would suggest to have: 
* override events: 10x base value
* events: 100x base value
like this we would expect the overrides to have a weight of 0.1 and normal events 1.0
regularization should then reduce the overrides to <0.1 and normal >0.5